### PR TITLE
:book: Fixed documentation

### LINKED ISSUE #264: Mismatch of config.toml with git checkout and image in podman compose:   pydantic_core._pydantic_core.ValidationError: 1 validation error for KaiConfig -  Unable to extract tag using discriminator 'provider' [type=union_tag_not_found, input_value={'solution_detectors': 'n...'password': 'dog8code'}}, input_type=dict]
I think the below is due to a mismatch where we are tagging images to isolate all changes, but we allow the config to be modified and used from the local checkout.

So in my situation:'
 -> running `podman compose up` that uses a `:stable` tag pointing to:  [commit 65a5761d479d84c15ccf0769e2bf00ea71a4836c](https://github.com/konveyor/kai/commit/65a5761d479d84c15ccf0769e2bf00ea71a4836c)
 -> My kai checkout of 'main' is at [commit 11a28d32992d9888234c2ea3a630a52733a48842](https://github.com/konveyor/kai/commit/11a28d32992d9888234c2ea3a630a52733a48842)

With this combination, the code running inside of the :stable image is reading the local checkout to parse the kai/config.toml and its crashing as I suspect we changed structure of kai/config.toml

Error from `podman compose up`

```
kai-1     |   File "/kai/kai/server.py", line 265, in app
kai-1     |     config = KaiConfig.model_validate_filepath(
kai-1     |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
kai-1     |   File "/kai/kai/models/kai_config.py", line 125, in model_validate_filepath
kai-1     |     return KaiConfig.model_validate(model_dict)
kai-1     |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
kai-1     |   File "/opt/app-root/lib64/python3.12/site-packages/pydantic/main.py", line 551, in model_validate
kai-1     |     return cls.__pydantic_validator__.validate_python(
kai-1     |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
kai-1     | pydantic_core._pydantic_core.ValidationError: 1 validation error for KaiConfig
kai-1     | incident_store
kai-1     |    pydantic_core._pydantic_core.ValidationError: 1 validation error for KaiConfig
kai-1     |     For further information visit https://errors.pydantic.dev/2.7/v/union_tag_not_found
```

https://gist.github.com/jwmatthews/2619140e479c4029b5e90d9639d41a68

The workaround for now is:
- `git checkout stable`
- `podman compose up`

This problem will fade away when we bump the stable tag to match main, yet there is potential for it to reoccur as long as we allow a relationship of the container image reading local disk for kai/config.toml... in theory the config file does not change often so the problem isn't expected to occur a lot, but its a potential breakage we may want to isolate.

### LINKED ISSUE #287: KeyError: 'GENAI_KEY' - We require GENAI_KEY to be present even with demo_mode
Running with demo_mode true and podman compose up we still do a check and require GENAI_KEY to be in the env

```
kai_1               | Config loaded: KaiConfig(log_level='info', demo_mode=True, trace_enabled=True, incident_store=KaiConfigIncidentStorePostgreSQL(provider='postgresql', args=KaiConfigIncidentStorePostgreSQLArgs(host='kai_db', database='kai', user='kai', password='dog8code')), models=KaiConfigModels(provider='ChatIBMGenAI', args={'model_id': 'meta-llama/llama-3-70b-instruct', 'parameters': {'max_new_tokens': 2048}}, template=None, llama_header=None))
kai_1               | Logging for KAI has been initialized and the level set to INFO
kai_1               | INFO - 2024-08-09 14:52:44,897 - [           server.py:291  -                  app()] - Tracing of actions is enabled
kai_1               | INFO - 2024-08-09 14:52:44,897 - [           server.py:296  -                  app()] - DEMO_MODE is enabled. LLM responses will be cached
kai_1               | INFO - 2024-08-09 14:52:44,901 - [             psql.py:55   -             __init__()] - Connected to the PostgreSQL server.
kai_1               | INFO - 2024-08-09 14:52:44,903 - [           server.py:299  -                  app()] - Selected incident store: postgresql
kai_1               | [2024-08-09 14:52:44 +0000] [24] [ERROR] Exception in worker process
kai_1               | Traceback (most recent call last):
kai_1               |   File "/opt/app-root/lib64/python3.12/site-packages/gunicorn/arbiter.py", line 609, in spawn_worker
kai_1               |     worker.init_process()
kai_1               |   File "/opt/app-root/lib64/python3.12/site-packages/aiohttp/worker.py", line 51, in init_process
kai_1               |     super().init_process()
kai_1               |   File "/opt/app-root/lib64/python3.12/site-packages/gunicorn/workers/base.py", line 134, in init_process
kai_1               |     self.load_wsgi()
kai_1               |   File "/opt/app-root/lib64/python3.12/site-packages/gunicorn/workers/base.py", line 146, in load_wsgi
kai_1               |     self.wsgi = self.app.wsgi()
kai_1               |                 ^^^^^^^^^^^^^^^
kai_1               |   File "/opt/app-root/lib64/python3.12/site-packages/gunicorn/app/base.py", line 67, in wsgi
kai_1               |     self.callable = self.load()
kai_1               |                     ^^^^^^^^^^^
kai_1               |   File "/opt/app-root/lib64/python3.12/site-packages/gunicorn/app/wsgiapp.py", line 58, in load
kai_1               |     return self.load_wsgiapp()
kai_1               |            ^^^^^^^^^^^^^^^^^^^
kai_1               |   File "/opt/app-root/lib64/python3.12/site-packages/gunicorn/app/wsgiapp.py", line 48, in load_wsgiapp
kai_1               |     return util.import_app(self.app_uri)
kai_1               |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
kai_1               |   File "/opt/app-root/lib64/python3.12/site-packages/gunicorn/util.py", line 424, in import_app
kai_1               |     app = app(*args, **kwargs)
kai_1               |           ^^^^^^^^^^^^^^^^^^^^
kai_1               |   File "/kai/kai/server.py", line 301, in app
kai_1               |     webapp["model_provider"] = ModelProvider(config.models)
kai_1               |                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
kai_1               |   File "/kai/kai/model_provider.py", line 50, in __init__
kai_1               |     "client": Client(credentials=Credentials.from_env()),
kai_1               |                                  ^^^^^^^^^^^^^^^^^^^^^^
kai_1               |   File "/opt/app-root/lib64/python3.12/site-packages/genai/credentials.py", line 54, in from_env
kai_1               |     api_key=os.environ[api_key_name or "GENAI_KEY"],
kai_1               |             ~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
kai_1               |   File "<frozen os>", line 685, in __getitem__
kai_1               | KeyError: 'GENAI_KEY'
```

We want folks who are evaluating Kai to be able to run in demo_mode without needing to worry about API Keys.

### PR DESCRIPTION
**Pull request by JonahSussman**
Make sure to build the podman image with `podman build -f build/Containerfile -t my_kai .` and then run compose with `TAG=latest IMAGE=localhost/my_kai podman compose up`

Issues fixed:
- Fixes #264
- Fixes #287

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,100 +1,171 @@
-# Konveyor AI (Kai)
+# Kai (Konveyor AI)
```
Comment (jwmatthews) [inline README.md:1]: Note, I do not have a strong preference in either direction.  Just wanted to raise question if this is how we want to introduce Kai.

I was under the impression the recommended way to introduce an acronym was to write out the full name and place the acronym in parentheses.

File: docs/OpenAI-API-Compatible-Alternatives.md
Comment (jwmatthews) [inline docs/OpenAI-API-Compatible-Alternatives.md:1]: I'm unsure why this was deleted.
Are the contents moved elsewhere?

File: README.md [hunk 2]
```diff
@@ -1,100 +1,171 @@
...
+<div align="center">
+  <a href="https://github.com/konveyor/kai">
+    <img src="docs/images/kai_logo.png" alt="KAI Logo" width="200" height="200">
+  </a>
+</div>
+<!-- trunk-ignore-emd -->
 
-Pronunciation of 'kai': https://www.howtopronounce.com/ka%C3%AC-4
+Kai [/paɪ/ (rhymes with pie)](https://www.howtopronounce.com/ka%C3%AC-4) - An
+awesome tool to accelerate application modernization, supercharged with
```
[the remark this answers is not in the document]
Comment (JonahSussman) [reply README.md:12]: This was my attempt was to make it more inviting. We can tone it down if need be

File: docs/OpenAI-API-Compatible-Alternatives.md
Comment (JonahSussman) [reply docs/OpenAI-API-Compatible-Alternatives.md:1]: I merged this file with the llm selection doc, so it's present in `docs/llm_selection.md`. To me, this just made sense as someone will looking to use a specific llm will already be guided to that document

File: README.md (hunk 1 above)
Comment (JonahSussman) [reply README.md:1]: I think we’ve been consistently referring to it as "Kai" throughout the project, so I decided to lead with that here as well. While "Konveyor AI" provides context, we don’t typically use that name anywhere else in our documentation or discussions. By introducing it this way, I think we maintain alignment with how we refer to the project internally and externally.

File: README.md [hunk 3]
```diff
@@ -1,100 +1,178 @@
...
+### Guided Walk-through
+
+After you have the kai backend running via `podman compose up` you can run
+through a guided scenario we have to show Kai in action at
+[docs/scenarios/demo.md](docs/scenarios/demo.md). This document walks through a
+guided scenario of using Kai to complete a migration of a Java EE app to
+Quarkus.
 
-## Technical design documents
+### Others ways to run Kai
```
Comment (jwmatthews) [inline README.md:150]: Typo from earlier commit.

'Others' => 'Other'

File: docs/getting_started.md [hunk 4]
```diff
@@ -1,38 +1,61 @@
...
 
-- _The easiest way to run is to leverage prebuilt container images we publish to [quay.io/konveyor/kai](https://quay.io/repository/konveyor/kai?tab=tags), you can learn more about early builds at [docs/Evaluation_Builds.md](/docs/Evaluation_Builds.md)_
-- _This is the simplest configuration which will limit configuration choices and will use cached LLM results so that you may evaluate Kai without having your own API Keys_
-  - The cached data uses a `KAI__DEMO_MODE=TRUE` mode for running the backend. See [docs/contrib/Demo_Mode.md](/docs/contrib/Demo_Mode.md) for more information.
-    - Configured via:
-      - [kai/config.toml](/kai/config.toml): `demo_mode = true`
-      - [compose.yaml](/compose.yaml): `KAI__DEMO_MODE: "TRUE"`
-  - Follow the guided scenario at [docs/scenarios/demo.md](/docs/scenarios/demo.md) to evaluate Kai
+The easiest way to run Kai is to leverage the prebuilt container images
+wepublish to
```
Comment (jwmatthews) [inline docs/getting_started.md:38]: Missing a space 'wepublish'

**Pull request merged**